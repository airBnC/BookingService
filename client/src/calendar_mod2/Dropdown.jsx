import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import CheckInCalendar from './CheckInCalendar';
import CheckOutCalendar from './CheckOutCalendar';
import arrow from '../images/arrow.png';

export const Holder = styled.div`
  position: absolute;
  z-index: 2;
  display: flex;
  flex-direction: column;
  width: 286px;
  border: 1px solid rgb(172, 172, 172);
  border-radius: 1%;
  padding: 20px;
  background-color: white;
`;

export const Arrow = styled.div`
  border-radius: 10%;
  content: url(${arrow});
  height: 20px;
  border: 1px solid rgb(172, 172, 172);
  border-radius: 10%;
  padding: 3px;
  transform: ${props => (props.left ? 'rotate(0.5turn)' : '')};
    &:active{
      background-color: #79CCCD;
    }
    &:hover {
      cursor: pointer;
    }
`;

export const SectionOne = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-top: 10px;
  font-size: 18px;
  font-weight: bold;
`;

export const SectionTwo = styled.div`
  display: flex;
  align-items: center;
`;

export const SectionThree = styled.div`
  font-size: 12px;
  margin-top: 12px;
  display: flex;
  flex-direction: column;
`;

export default function Dropdown(props) {
  let calendar;
  let holderArrow;
  if (props.clicked === 1) {
    calendar = (<CheckInCalendar {...props} />);
    holderArrow = (
      <svg height="10" width="250" fill="rgb(172, 172, 172)" stroke="rgb(172, 172, 172)" position="absolute">
        <path d="M 15,25 35, 25 25, 0 Z" height="0px" position="absolute" />
      </svg>);
  } else if (props.clicked === -1) {
    calendar = (<CheckOutCalendar {...props} />);
    holderArrow = (
      <svg height="10" width="250" fill="rgb(172, 172, 172)" stroke="rgb(172, 172, 172)">
        <path d="M 0,25 35, 25 25, 15 Z" height="0px" />
      </svg>);
  }
  if (props.clicked === 1 || props.clicked === -1) {
    return (
      <div>
        {holderArrow}
        <Holder>
          <SectionOne>
            <Arrow left />
            <div> {props.month} {props.year} </div>
            <Arrow />
          </SectionOne>
          <SectionTwo>
            {calendar}
          </SectionTwo>
          <SectionThree>
            <div> {props.minStay} day minimum. </div>
            <div> Updated 2 days ago </div>
          </SectionThree>
        </Holder>
      </div>
    );
  }
  return null;
}

Dropdown.propTypes = {
  clicked: PropTypes.number,
  month: PropTypes.string,
  year: PropTypes.number,
  dateClick: PropTypes.func,
  // dates: PropTypes.arrayOf( PropTypes.shape( {
  //   date: PropTypes.date,
  // } ) ),

};
Dropdown.defaultProps = {
  clicked: 0,
  month: 'June',
  year: 2018,
  dateClick: () => null,
  // dates: [],
};
