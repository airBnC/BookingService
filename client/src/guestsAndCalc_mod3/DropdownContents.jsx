import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const GuestHolder = styled.div`
  display: flex;
  flex-direction: row;
`;

const GuestDetails = styled.div`
`;

const DownButton = styled.button`
`;

const UpButton = styled.button`
`;

export default function DropdownContents( props ) {
  DropdownContents.propTypes = {
    adults: PropTypes.number,
    childrens: PropTypes.number,
    infants: PropTypes.number,
    btnClick: PropTypes.func,
  };
  DropdownContents.defaultProps = {
    adults: 0,
    childrens: 0,
    infants: 0,
    btnClick: () => null,
  };

  const thingsToRender = [ 'adults', 'children', 'infants' ];
  const toRender = thingsToRender.map( thing =>
    ( <GuestHolder>
      <div> {thing} </div>
      <DownButton />
      <div> {props.adults} </div>
      <UpButton />
      </GuestHolder> ) );

  return (
    <div>
      {toRender}
    </div>

  );
}
