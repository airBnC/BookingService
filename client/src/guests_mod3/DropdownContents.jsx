import React from 'react';
import PropTypes from 'prop-types';
import { Holder, FeatureHolder, Guest, GuestDetails, ButtonDown, ButtonUp, Number } from './DropdownContentsStyling';

const list = ['adults', 'children', 'infants'];
const desc = ['', 'Ages 2-12', 'Under 2'];

export default function DropdownContents(props) {
  const toRender = list.map((propName, index) => {
    const guest = propName.replace(/s|ren/g, '');
    const guestNum = props[guest];
    return (
      <Holder key={propName}>
        <FeatureHolder>
          <Guest guest={propName}> {propName} </Guest>
          <GuestDetails> {desc[index]} </GuestDetails>
        </FeatureHolder>
        <FeatureHolder button guest={propName}>
          <ButtonDown
            disabled={(guest === 'adult' && guestNum === 1) || guestNum === 0}
            onClick={() => props.btnClick(propName, -1)}
            number={guestNum}
            guest={propName}
          >-
          </ButtonDown>
          <Number> {guestNum} </Number>
          <ButtonUp
            disabled={(guest === 'infant' && guestNum === 5) || (props.totalGuests === props.maxGuests && (guest === 'adult' || guest === 'child'))}
            onClick={() => props.btnClick(propName, 1)}
            number={guestNum}
            total={props.totalGuests}
            max={props.maxGuests}
            guest={propName}
          >+
          </ButtonUp>
        </FeatureHolder>
      </Holder>
    );
  });

  return (
    <div>
      {toRender}
    </div>

  );
}

DropdownContents.propTypes = {
  // btnClick: PropTypes.func,
  // adult: PropTypes.number,
  // child: PropTypes.number,
  // infant: PropTypes.number,
  totalGuests: PropTypes.number,
  maxGuests: PropTypes.number,

};
DropdownContents.defaultProps = {
  // btnClick: () => null,
  // adult: 0,
  // child: 0,
  // infant: 0,
  totalGuests: 0,
  maxGuests: 0,
};
