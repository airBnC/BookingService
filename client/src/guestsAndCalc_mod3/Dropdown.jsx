import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import DropdownContents from './DropdownContents';

const Holder = styled.div`
  display: flex;
  flex-direction: column;
  width: 230px;
  border: 1px solid rgb(172, 172, 172);
  padding: 10px;
  padding-top: 0px;
`;

const Details = styled.div`
  font-size: 12px;
  margin-top: 10px;
`;

export default function Dropdown( props ) {
  Dropdown.propTypes = {
    clicked: PropTypes.bool,
    maxGuests: PropTypes.number,
    adult: PropTypes.number,
    child: PropTypes.number,
    infant: PropTypes.number,
    btnClick: PropTypes.func,
  };
  Dropdown.defaultProps = {
    clicked: false,
    maxGuests: 0,
    adult: 0,
    child: 0,
    infant: 0,
    btnClick: () => null,
  };
  if ( props.clicked ) {
    console.log( 'I was clicked!' );
    return (
      <Holder>
        <DropdownContents
          adult={props.adult}
          child={props.child}
          infant={props.infant}
          btnClick={props.btnClick}
        />
        <Details> {props.maxGuests } guests maximum. Infants don't count toward the number of guests. </Details>
      </Holder>
    );
  }
  return null;
}
