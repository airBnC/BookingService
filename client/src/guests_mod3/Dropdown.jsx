import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import DropdownContents from './DropdownContents';
import { DropdownHolder, Close } from '../ModStylings';

export const Details = styled.div`
  font-size: 12px;
  margin-top: 12px;
`;

export default function Dropdown(props) {
  if (props.clicked) {
    return (
      <DropdownHolder guest>
        <DropdownContents
          {...props}
        />
        <Details>
          {props.maxGuests } guests maximum.
          Infants donâ€™t count toward the number of guests.
        </Details>
        <Close onClick={props.close}> Close </Close>
      </DropdownHolder>
    );
  }
  return null;
}

Dropdown.propTypes = {
  clicked: PropTypes.bool,
  btnClick: PropTypes.func,
  close: PropTypes.func,
  adult: PropTypes.number,
  child: PropTypes.number,
  infant: PropTypes.number,
  totalGuests: PropTypes.number,
  maxGuests: PropTypes.number,

};
Dropdown.defaultProps = {
  clicked: false,
  btnClick: () => null,
  close: () => null,
  adult: 0,
  child: 0,
  infant: 0,
  totalGuests: 0,
  maxGuests: 0,
};
