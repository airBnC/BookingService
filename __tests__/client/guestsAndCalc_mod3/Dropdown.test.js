import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import 'jest-styled-components';
import { configure, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Dropdown, { Holder } from '../../../client/src/guestsAndCalc_mod3/Dropdown';
import DropdownContents from '../../../client/src/guestsAndCalc_mod3/DropdownContents';

configure( { adapter: new Adapter() } );

const props = {
  clicked: false,
  btnClick: () => null,
  close: () => null,
  adult: 0,
  child: 0,
  infant: 0,
  totalGuests: 0,
  maxGuests: 0,
};

describe( 'Holder Component', () => {
  it( 'is the main flex component', () => {
    const tree = renderer
      .create( <Holder> things inside </Holder> )
      .toJSON();
    expect( tree ).toHaveStyleRule( 'display', 'flex' );
    expect( tree ).toMatchSnapshot();
  } );
} );
describe( 'Dropdown Component', () => {
  it( 'renders the right amount of children', () => {
    const wrapper = shallow( <Dropdown {...props} /> );
    expect( wrapper.find( renderer.create( <DropdownContents /> ) ).children() ).toHaveLength( 3 );
  } );
} );
