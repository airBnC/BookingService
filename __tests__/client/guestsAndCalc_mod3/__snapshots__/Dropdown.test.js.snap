// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Holder Component is the main flex component 1`] = `
.c0 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 230px;
  border: 1px solid rgb(172,172,172);
  margin-top: 50px;
  padding: 10px;
  padding-top: 0px;
  background-color: white;
}

<div
  className="c0"
>
   things inside 
</div>
`;
